lib/carto/tree/filterset.js

```diff
@@ -92,6 +92,9 @@ tree.Filterset.prototype.toJS = function(env) {
       val = filter._val.toString(true);
     }
     var attrs = "data";
+    if (op === '=~') {
+      return "(typeof " + attrs + "['" + filter.key.value  + "'] === 'string') && " + attrs + "['" + filter.key.value  + "'].match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'") + "'" : val) + ")";
```

Comment:
Either that or cast `data[filter.key.value]` as a string, as the problem is to use `.match` on a Number. If you test it, you can see that is possible to use `=~` on numbers.